src/sqlparser/src/parser.rs

```diff
@@ -2001,6 +2025,17 @@ impl Parser {
         }
     }
 
+    pub fn parse_optional_watermark(&mut self) -> Result<Option<SourceWatermark>, ParserError> {
+        if self.parse_keywords(&[Keyword::WATERMARK, Keyword::FOR]) {
```

Comment:
Maybe we should expect `FOR` after `WATERMARK` for a more straightforward error message.

Reply:
Good catch!